zkevm-circuits/src/keccak_circuit/keccak_packed.rs

```diff
@@ -0,0 +1,1770 @@
+                        - 2.expr() * round_cst_expr.clone();
+                    let fat = split(
+                        meta,
+                        &mut cell_values,
+                        &mut cb,
+                        input,
+                        0,
+                        part_size_ext,
+                        false,
+                    );
+                    os[i][j] = decode(transform(
+                        "chi ext",
+                        meta,
+                        &mut cell_values,
+                        &mut lookup_counter,
+                        fat.clone(),
+                        chi_ext_table,
+                    ));
+                    cb.require_equal("next row check", os[i][j].clone(), s_next[i][j].clone());
+                } else {
+                    let mut fat = Vec::new();
+                    if get_mode() {
+                        for ((part_a, part_b), part_c) in os_parts[i][j]
+                            .iter()
+                            .zip(os_parts[(i + 1) % 5][j].iter())
+                            .zip(os_parts[(i + 2) % 5][j].iter())
+                        {
+                            //println!("part size: {}", part.num_bits);
+                            let expr = pack_bit(3, part_size_base) + part_b.expr.clone()
+                                - 2.expr() * part_a.expr.clone()
+                                - part_c.expr.clone();
+                            fat.push(Part {
+                                num_bits: part_size_base,
+                                expr: expr.clone(),
+                                parts: vec![expr.clone()],
+                            });
+                        }
+                    } else {
+                        let input = pack_bit(3, 64) + s[(i + 1) % 5][j].clone()
+                            - 2.expr() * s[i][j].clone()
+                            - s[(i + 2) % 5][j].clone();
+                        fat = split(
+                            meta,
+                            &mut cell_values,
+                            &mut cb,
+                            input,
+                            0,
+                            part_size_base,
+                            false,
+                        );
+                    }
+                    os[i][j] = decode(transform(
+                        "chi base",
+                        meta,
+                        &mut cell_values,
+                        &mut lookup_counter,
+                        fat.clone(),
+                        chi_base_table,
+                    ));
+                    cb.require_equal("next row check", os[i][j].clone(), s_next[i][j].clone());
+                }
+            }
+        }
+        println!("- Post chi:");
+        println!("Lookups: {}", lookup_counter);
+        println!("Columns: {}", cell_values.len());
+        println!("num_parts_pre: {}", num_parts_pre);
+        println!("num_parts_post: {}", num_parts_post);
```

Comment:
For sure, just there because it's very helpful while developing/optimizing to keep track of how many cells/lookups are required for the different steps.